### ISSUE TITLE
CookieContainer do not handle session cookies (when Expires is not specified)

### ISSUE THREAD

**Issue by M-Boukhlouf** [dotnet/corefx#40934]
# CookieContainer do not handle session cookies (when Expires is not specified)

According to this page: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

> The maximum lifetime of the cookie as an HTTP-date timestamp. If not specified, the cookie will have the lifetime of a session cookie. A session is finished when the client is shut down meaning that session cookies will get removed at that point.

Yet, in this case when the Expires is not specified in this Set-Cookie header:
`Set-Cookie: name=value; path=/store; httponly`
The CookieContainer set the Expires property to "1/1/0001 12:00:00 AM"
Which is expired and the cookie is not used in the following requests.

**Comment by davidsh** [comment]
.NET Cookies don't work exactly the way browsers handle them.  There is no such thing as "session cookies" in .NET.  Cookies are received on the wire or created by APIs. Then they are stored in an in-memory CookieContainer object. If there is no 'Expires' attribute, then they live "forever" in the sense of living in the CookieContainer object. But when the application exits (the one that created the CookieContainer), the cookie goes away.

**Comment by davidsh** [comment]
@scalablecory fyi. "Cookie" is part of System.Net namespace.

**Comment by M-Boukhlouf** [comment]
> If there is no 'Expires' attribute, then they live "forever" in the sense of living in the CookieContainer object. But when the application exits (the one that created the CookieContainer), the cookie goes away.

False, the whole issue I'm having is that the cookie is treated as if it's already expired, thus it's not used afterwards.

**Comment by alnikola** [comment]
I tried reproducing this issue by adding a cookie header without Expires property to a CookieContainer, and then retrieving it, however everything worked **properly**. I was able to successfully retrieve such cookie and Expired property was false. Please, check the following code sample. It is correct that Expires is set to DateTime.MinValue in that case, but we treat this as "infinite" expiration. [The special check is implemented in Expired property](https://github.com/dotnet/runtime/blob/6b6d9135298500a753b3393b1c7cc4aa94617aa2/src/libraries/System.Net.Primitives/src/System/Net/Cookie.cs#L197).
Could you please provide a code sample reproducing this issue?
This test **succeeds**.
```C#
CookieContainer cc = new CookieContainer();
Uri uri = new Uri("http://localhost");
cc.SetCookies(uri, "name98=value98");
Cookie actualCookie = cc.GetCookies(uri).First();
Assert.Equal("name98", actualCookie.Name);
Assert.Equal("value98", actualCookie.Value);
Assert.False(actualCookie.Expired);

**Comment by M-Boukhlouf** [comment]
> I tried reproducing this issue by adding a cookie header without Expires property to a CookieContainer, and then retrieving it, however everything worked **properly**. I was able to successfully retrieve such cookie and Expired property was false. Please, check the following code sample. It is correct that Expires is set to DateTime.MinValue in that case, but we treat this as "infinite" expiration. [The special check is implemented in Expired property](https://github.com/dotnet/runtime/blob/6b6d9135298500a753b3393b1c7cc4aa94617aa2/src/libraries/System.Net.Primitives/src/System/Net/Cookie.cs#L197).
> Could you please provide a code sample reproducing this issue?
> This test **succeeds**.
> 
> ```cs
> CookieContainer cc = new CookieContainer();
> Uri uri = new Uri("http://localhost");
> cc.SetCookies(uri, "name98=value98");
> Cookie actualCookie = cc.GetCookies(uri).First();
> Assert.Equal("name98", actualCookie.Name);
> Assert.Equal("value98", actualCookie.Value);
> Assert.False(actualCookie.Expired);
> ```

But in my case, the CookieContainer was in use by a HttpClient object. I'm sure that the cookie will be not be sent in the next request, and just after the request, it'll be shown as Expired.

**Comment by alnikola** [comment]
HttpConnection [calls GetCookieHeader method](https://github.com/dotnet/runtime/blob/6afe96cc2e950379d13a87efa1fa5a651cbce9bd/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L389) to attach cookies to a request if **UseCookies is set to true**. GetCookieHeader internally calls the same logic as GetCookies, thus the expiration behavior is the same, meaning cookies with the default Expires value DateTime.MinValue will have Expired returning **false** and will be **attached** to the request.

**Comment by scalablecory** [comment]
@M-Boukhlouf can you provide some repro code using `HttpClient` that demonstrates the issue?
